src-input/duk_bi_buffer.c

```diff
@@ -1196,7 +1197,18 @@ DUK_INTERNAL duk_ret_t duk_bi_nodejs_buffer_tostring(duk_hthread *thr) {
 	}
 	DUK_HBUFOBJ_ASSERT_VALID(h_this);
 
-	/* Ignore encoding for now. */
+	/* TODO: support other encodings.  currently only 'utf8' is supported. */
+	if (duk_is_undefined(thr, 0)) {
+		encoding = "utf8";
+	} else if (duk_is_string(thr, 0)) {
+		encoding = duk_to_string(thr, 0);
+		DUK_ASSERT(duk_is_string(thr, 0));
+		if(DUK_STRCMP(encoding, "utf8") != 0) {
+			DUK_DCERROR_TYPE_INVALID_ARGS(thr);
+		}
+	} else {
+		DUK_DCERROR_TYPE_INVALID_ARGS(thr);
+	}
```

Comment:
This block might be replaced by:
```
encoding = duk_opt_string(thr, 0, "utf8");
if (DUK_STRCMP(encoding, "utf8") != 0) {
        DUK_DCERROR_TYPE_INVALID_ARGS(thr);
}
```

duk_opt_string() should:
- Accept a string
- Return "utf8" when the value is undefined or the index is out-of-range
- TypeError on anything else, including a `null`

Reply:
The logic has been factored out since the same check is required for `Buffer.isEncoding`.